nilearn/image/image.py

```diff
@@ -984,7 +986,7 @@ def threshold_img(
         img_data = img_data[:, :, :, 0]
 
     # Reconstitute img object
-    thresholded_img = new_img_like(img, img_data, affine)
+    thresholded_img = new_img_like(img, img_data, affine, copy_header=True)
```

Comment:
Given that this will go through a deprecation cycle for this you want to leave the current behavior unchanged and let user opt-in until we finally deprecate it.

So you need to change the function signature and the doc string too:

```python
def threshold_img(
    img,
    threshold,
    cluster_threshold=0,
    two_sided=True,
    mask_img=None,
    copy=True,
    copy_header=False # add it here
)
"""
updated doc string with description of the new parameter.

    copy_header: :obj:`bool`, default = False

        explain what this does

        .. versionadded:: 0.8.1

"""

...

   # just pass the value here
   thresholded_img = new_img_like(img, img_data, affine, copy_header=copy_header)
```

Reply:
Only commenting on this function but this should apply to all pubic functions that gets changed

Reply:
Ok, that makes sense. I was thinking of doing that in #4427.

So IIUC, in release 0.13.0, we'll simply set the default to True and remove the warnings, right?

Reply:
> So IIUC, in release 0.13.0, we'll simply set the default to True and remove the warnings, right?

yes exactly 👍🏾